### PR TITLE
Move out from pbr

### PR DESCRIPTION
**Pull request by sileht**
All pbr interesting features have been implemented in setuptools since
version 30.3.0 (8 Dec 2016).
Also pbr will break us soon and force us to found a new solution
to generate things during setup.py steps.
https://github.com/gnocchixyz/gnocchi/issues/755

When you use pbr you cannot override some cmdclass. So we cannot
make our hooks working anymore.

This change also fixes the header of gnocchi-api.

This may impact downstream packaging as setuptools versions are a bit
old:
* xenial: 20.7
* zesty: 33.1.1
* centos7/rhel7: 0.9.8 ... really ?
* rdo queen: 22.0
* rdo rocky plan to use >= 38.6

Closes-bug: #755

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now d6b98a725f3e0d605625ab08f98c724bb319d723

**Comment by sileht** [comment]
Pull-request updated, HEAD is now cae1460fb9df0b89eed42a447b8156b948b2fff8

**Comment by sileht** [comment]
Pull-request updated, HEAD is now ad0177073f9c1ba753a3ec9a1654f04c344a2bcf

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 6b06335df48d1e4affa04a2feafacd751610c5bf

**Comment by sileht** [comment]
Pull-request updated, HEAD is now d53b21579049871cf1e297f7d7ca8ae128698185

**Comment by sileht** [comment]
Pull-request updated, HEAD is now ec2f714a6b3c33473a8a0dd495c0b8ce4687551a

**Comment by sileht** [comment]
Pull-request updated, HEAD is now b6ec4e7721c2eb3a474369037b6d46848690987a

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 3f26e29b6d7816e9e1a78cc34b02d0d309528b5a

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 2cb99eec3c162cda514afb5cc6c4687571505999

**Comment by sileht** [comment]
Pull-request updated, HEAD is now a4e54eaf42f9c350f91fd0c0f84a69a2697540e0

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 524aa9024bb59e7c8d8831f477aac36cec550233

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 30c975d7e7435bbc299a79f00240618f49dbb492

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 0f1ae1e478bc6093a70aeba8e31066ff74dc94d9

**Comment by sileht** [comment]
Pull-request updated, HEAD is now e73115f0152e0f7f3006eb05086081dcc2d1f495

File: tox.ini [hunk 1]
```diff
@@ -145,6 +148,8 @@ setenv = GNOCCHI_STORAGE_DEPS=file
          GNOCCHI_TEST_DEBUG=1
 deps = {[testenv:docs]deps}
        sphinxcontrib-versioning
+# for <= 4.3 doc
```
Comment (jd) [inline tox.ini]: < 4.3

File: gnocchi/setuptools.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import subprocess
+
+from distutils import version
+from setuptools.command import develop
+from setuptools.command import easy_install
+from setuptools.command import egg_info
+from setuptools.command import install_scripts
+
+
+SCRIPT_TMPL = """
```
Comment (jd) [inline gnocchi/setuptools.py]: do we really need a template for this? Can't we just ship this as a script?

**Comment by sileht** [comment]
Pull-request updated, HEAD is now e67230b569a2c0cf11c8a354da268c1231f25d8b

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 7bcff59b9eebb36e11e381859564b0d25b8d93e5

**Comment by sileht** [comment]
Pull-request updated, HEAD is now d7db3de7f40cacfbea9374ae8f32f8d8b51a93bc

File: gnocchi/setuptools.py (hunk 2 above)
Comment (sileht) [reply gnocchi/setuptools.py]: The current script have a wrong header (we hardcode python2), I use a tmpl to add the sheban with the correct python version, I ask setuptools/easy_install what the binary to use at install time, so the script will always work. Also we can't use an entry_point due to the wsgi stuff.

File: tox.ini (hunk 1 above)
Comment (sileht) [reply tox.ini]: done

File: MANIFEST.in [hunk 3]
```diff
@@ -1 +1,2 @@
-include etc/gnocchi/gnocchi.conf
+include ChangeLog
+include AUTHORS
```
Comment (jd) [inline MANIFEST.in]:
```
exclude .gitignore
exclude .github
```

File: setup.cfg [hunk 4]
```diff
@@ -72,21 +106,7 @@ test =
 test-swift =
     python-swiftclient
 
-[global]
-setup-hooks =
-    pbr.hooks.setup_hook
-
-[build_py]
-pre-hook.build_config = gnocchi.genconfig.prehook
```
Comment (jd) [inline setup.cfg]: Now the final tarball does not have `gnocchi.conf`. Not sure it's a big deal since we have a default config generator. Just saying.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 85e85950c0999404b76d863943ab8f3dffb4b475

File: MANIFEST.in (hunk 3 above)
Comment (sileht) [reply MANIFEST.in]: added

File: setup.cfg (hunk 4 above)
Comment (sileht) [reply setup.cfg]: I have put it back.

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 3ee61c1d493f5c858d8630cb7d12107ffd229fac

**Comment by sileht** [comment]
Pull-request updated, HEAD is now eaf4d0d7b3259e1a3758e47578c4c8a92f6f26ab

**Comment by sileht** [comment]
I finally removed the etc/gnocchi/gnocchi.conf from the tarball. Because it requires all dependencies installed to be generated at sdist time, and is already missing in the tarball on pypi.

File: tox.ini [hunk 5]
```diff
@@ -146,6 +148,8 @@ setenv = GNOCCHI_STORAGE_DEPS=file
          GNOCCHI_TEST_DEBUG=1
 deps = {[testenv:docs]deps}
        sphinxcontrib-versioning
+# for < 4.3 doc
+       pbr
```
Comment (chungg) [inline tox.ini]: do we need to upper cap pbr?

Comment (sileht) [reply tox.ini]: I don't think so

Comment (chungg) [reply tox.ini]: kk, just asking since it says new pbr will break us.

**Pull request merged**